src/eckit/types/FloatCompare.cc

```diff
@@ -1,35 +1,23 @@
 // #include <cmath>
 
-// Some of the math.h/cmath functions are not clean when switching to C++11
-#if __cplusplus <= 199711L
-#include <cmath>
-#else
-#include <cmath>
-#define fpclassify(x) std::fpclassify((x))
-#define isinf(x) std::isinf((x))
-#define isnan(x) std::isnan((x))
-#define signbit(x) std::signbit((x))
-#endif
-
 #include <sys/types.h>
+#include <cmath>
 #include <limits>
 
 #include "eckit/exception/Exceptions.h"
 #include "eckit/types/FloatCompare.h"
 
+using std::fpclassify;
```

Comment:
This seems unrelated.

We weren't using the std:: ones as we had issues on some compiler combinations. Not sure I want to couple this change in this context?

Reply:
we included the `eckit/format/Format.h` in `eckit/log/Log.h` - this was then included in `FloatCompare`

Here some macros to math functions were used to avoid having to prefix `std::` ... however as they were declared as macros, they caused a lot of conflicts with the included libtfmt headers.

In that sense using the `using std::function;` approach is much cleaner. I'm not sure if this approach was properly tested before? Moreover we are now using C++17 - I'd be really supprised if there are still compilers complaining about these things. Also the CI passes.